pkg/cloud/cloud.go

```diff
@@ -638,6 +642,58 @@ func (c *cloud) CreateDisk(ctx context.Context, volumeName string, diskOptions *
 	return &Disk{CapacityGiB: size, VolumeID: volumeID, AvailabilityZone: zone, SnapshotID: snapshotID, OutpostArn: outpostArn}, nil
 }
 
+// execBatchDescribeVolumesModifications executes a batched DescribeVolumesModifications API call
+func execBatchDescribeVolumesModifications(svc EC2API, input []string) (map[string]*types.VolumeModification, error) {
+	klog.V(7).InfoS("execBatchDescribeVolumeModifications", "volumeIds", input)
+	request := &ec2.DescribeVolumesModificationsInput{
+		VolumeIds: input,
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), batchDescribeTimeout)
+	defer cancel()
+
+	resp, err := describeVolumesModifications(ctx, svc, request)
+	if err != nil {
+		return nil, err
+	}
+
+	result := make(map[string]*types.VolumeModification)
+
+	for _, i := range resp {
+		volumeModification := i
+		result[*volumeModification.VolumeId] = &volumeModification
+	}
+
+	klog.V(7).InfoS("execBatchDescribeVolumeModifications: success", "result", result)
+	return result, nil
+}
+
+// batchDescribeVolumesModifications processes a DescribeVolumesModifications request by queuing the task and waiting for the result.
+func (c *cloud) batchDescribeVolumesModifications(request *ec2.DescribeVolumesModificationsInput) (*types.VolumeModification, error) {
+	var task string
+
+	if len(request.VolumeIds) == 1 && request.VolumeIds[0] != "" {
+		task = request.VolumeIds[0]
+	} else {
+		return nil, fmt.Errorf("batchDescribeVolumesModifications: invalid request, request: %v", request)
+	}
+
+	ch := make(chan batcher.BatchResult[*types.VolumeModification])
+
+	b := c.bm.volumeModificationIDBatcher
+	b.AddTask(task, ch)
+
+	r := <-ch
+
+	if r.Err != nil {
+		return nil, r.Err
+	}
+	if r.Result == nil {
+		return nil, VolumeNotBeingModified
```

Comment:
Suggest adding a unit test for`r.Result == nil`

Reply:
On second thought, this block is not needed because if volume has no modifications an error will be returned. Will remove code block.